core/common/buffer.hpp

```diff
@@ -185,7 +185,7 @@ namespace fc::common {
 
     std::vector<uint8_t> &toVector();
 
-    inline operator std::vector<uint8_t> &&() && {
+    inline explicit operator std::vector<uint8_t> &&() && {
```

Comment:
this operator is also meant to be implicit, to avoid spelling "vector...".
also, "vector" member can be made public with 1 letter short name, to remove this operator.